x/migration/module/cmd/collect_morse_accounts.go

```diff
@@ -231,26 +232,44 @@ func collectInputApplicationStakes(inputState *migrationtypes.MorseStateExport,
 	for exportApplicationIdx, exportApplication := range inputState.AppState.Application.Applications {
 		appAddr := exportApplication.Address.String()
 
-		// DEV_NOTE: An account SHOULD exist for each actor.
 		if !morseWorkspace.hasAccount(appAddr) {
-			return ErrMorseExportState.Wrapf("account not found corresponding to application with address %q", appAddr)
-		}
-
-		appStakeAmtUpokt, ok := cosmosmath.NewIntFromString(exportApplication.StakedTokens)
-		if !ok {
-			return ErrMorseExportState.Wrapf("failed to parse application stake amount %q", exportApplication.StakedTokens)
+			logger.Logger.Warn().
+				Str("app_address", appAddr).
+				Msg("no account found for application")
+
+			// DEV_NOTE: If no auth account was found for this application, create a new one.
+			newAppAccount := &migrationtypes.MorseAuthAccount{
+				Type: "posmint/Account",
+				Value: &migrationtypes.MorseAccount{
+					Address: exportApplication.Address,
+					Coins:   []cosmostypes.Coin{},
+				},
+			}
+			if _, _, err := morseWorkspace.addAccount(appAddr, newAppAccount); err != nil {
+				return fmt.Errorf(
+					"adding application account to account balance of address %q: %w",
+					appAddr, err,
+				)
+			}
 		}
 
-		if err := morseWorkspace.addAppStake(appAddr, appStakeAmtUpokt); err != nil {
-			return fmt.Errorf(
-				"adding application stake amount to account balance of address %q: %w",
-				appAddr, err,
-			)
+		if exportApplication.StakedTokens != "" {
```

Comment:
1. Is there ever a situation where this could be empty? If so, #PUC
2. If not, should we log a warning, error or other?

Reply:
> 1. Is there ever a situation where this could be empty? If so, #PUC

It depends on Morse protocol details, so I'm not sure and haven't looked at the code to answer this question yet. Perhaps adding a warning log and running a recent snapshot would be sufficient to answer this question for practical purposes.

> 2. If not, should we log a warning, error or other?

I think this case is going to be problematic because of the constraints that we added with respect to account/actor claiming. I.e. unstaked, application, and supplier accounts can ONLY be claimed using their respective claim message, which currently precludes the possibility of an account which is staked as multiple actors.

Wdyt should be the correct behavior here?

Reply:
#PUC with a TODO_MAINNET_MIGRATION explaining what leads to this and we'll figure it out later.